### PR TITLE
LTDC peripheral

### PR DESCRIPTION
**Pull request by chengsun**
Significant parts of this peripheral API have been tested to work on a STM32F7508-DK Discovery kit.

(I created the `stm32f750` device to verify that this works. That new device depends on both this PR and https://github.com/stm32-rs/stm32-rs/pull/447. I'll create a PR for it once these have been merged.)

### CODE REVIEW

File: peripherals/ltdc/ltdc.yaml [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    DCGREEN: [0, 255]
+    DCBLUE: [0, 255]
+
+  "L[12]BFCR":
+    BF1:
+      Constant: [0b100, "BF1 = constant alpha"]
+      Pixel: [0b110, "BF1 = pixel alpha * constant alpha"]
+    BF2:
+      Constant: [0b100, "BF2 = 1 - constant alpha"]
+      Pixel: [0b110, "BF2 = 1 - pixel alpha * constant alpha"]
```
Comment (adamgreig) [inline peripherals/ltdc/ltdc.yaml:163]:
```suggestion
      Constant: [0b101, "BF2 = 1 - constant alpha"]
      Pixel: [0b111, "BF2 = 1 - pixel alpha * constant alpha"]
```

At least in RM0410 and RM0090, the BF2 values are different from BF1, and need 101 for constant alpha and 111 for pixel alpha.

**Review by adamgreig — CHANGES_REQUESTED** [review]
Thanks for the PR, this looks great! I've added a few fairly minor comments, otherwise I think this is good to merge.

File: peripherals/ltdc/ltdc.yaml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      ActiveHigh: [1, "Horizontal synchronization polarity is active high"]
+    VSPOL:
+      ActiveLow: [0, "Vertical synchronization polarity is active low"]
+      ActiveHigh: [1, "Vertical synchronization polarity is active high"]
+    DEPOL:
+      ActiveLow: [0, "Data enable polarity is active low"]
+      ActiveHigh: [1, "Data enable polarity is active high"]
+    PCPOL:
+      ActiveLow: [0, "Pixel clock polarity is active low"]
+      ActiveHigh: [1, "Pixel clock polarity is active high"]
```
[the remark this answers is not in the document]
Comment (chengsun) [reply peripherals/ltdc/ltdc.yaml:30]: I took this wording verbatim from RM0385, but I agree `RisingEdge` and `FallingEdge` make a lot more sense, given the clock is presumably edge-triggered, not level-triggered. Thanks!

File: peripherals/ltdc/ltdc.yaml [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  CDSR:
+    HSYNCS:
+      NotActive: [0, "Currently not in HSYNC phase"]
+      Active: [1, "Currently in HSYNC phase"]
+    VSYNCS:
+      NotActive: [0, "Currently not in VSYNC phase"]
+      Active: [1, "Currently in VSYNC phase"]
+    HDES:
+      NotActive: [0, "Currently not in horizontal Dtaa Enable phase"]
+      Active: [1, "Currently in horizontal Dtaa Enable phase"]
```
[the remark this answers is not in the document]
Comment (chengsun) [reply peripherals/ltdc/ltdc.yaml:108]: Thanks!

File: peripherals/ltdc/ltdc.yaml [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+      Clear: [1, "Clears the FUIF flag in the ISR register"]
+    CLIF:
+      Clear: [1, "Clears the LIF flag in the ISR register"]
+
+  LIPCR:
+    LIPOS: [0, 2047]
+
+  CPSR:
+    CXPOS: [0, 65535]
+    CYPOS: [0, 65535]
```
[the remark this answers is not in the document]
Comment (chengsun) [reply peripherals/ltdc/ltdc.yaml:97]: Ah, that makes sense. Thanks for explaining.

File: peripherals/ltdc/ltdc.yaml [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+      ActiveHigh: [1, "Data enable polarity is active high"]
+    PCPOL:
+      ActiveLow: [0, "Pixel clock polarity is active low"]
+      ActiveHigh: [1, "Pixel clock polarity is active high"]
+    DEN:
+      Disabled: [0, "Dither disabled"]
+      Enabled: [1, "Dither enabled"]
+    DRW: [0, 7]
+    DGW: [0, 7]
+    DBW: [0, 7]
```
[the remark this answers is not in the document]
Comment (chengsun) [reply peripherals/ltdc/ltdc.yaml:36]: Thank you!

File: peripherals/ltdc/ltdc.yaml (hunk 1 above)
Comment (chengsun) [reply peripherals/ltdc/ltdc.yaml:163]: Yes, in RM0385 as well. This was a copy-paste error, thank you for catching it!

**Comment by chengsun** [comment]
Thank you for the very thorough review!

**Comment by adamgreig** [comment]
I'm not sure what's happened exactly but it looks like the current changes in the PR are the same as originally and don't include any of the review changes?

**Comment by chengsun** [comment]
Really sorry about that. I primarily use Mercurial, and every time I switch back to git I forget to `git add` my changes first.

I've pushed the requested changes and rebased onto master now.

**Review by adamgreig — APPROVED** [review]
Thank you!

bors merge

**Pull request merged**